cfme/fixtures/v2v_fixtures.py

```diff
@@ -161,6 +163,92 @@ def __vddk_url(): # noqa
     return url
 
 
+def __conversion_data_rhev(target_provider):  # noqa
+    resource_type = "ManageIQ::Providers::Redhat::InfraManager::Host"
+    engine_key = conf.credentials[target_provider.data["ssh_creds"]]
+    auth_user = engine_key.username
+    ssh_client = ssh.SSHClient(
+        hostname=target_provider.hostname,
+        username=engine_key.username,
+        password=engine_key.password,
+    )
+    private_key = ssh_client.run_command("cat /etc/pki/ovirt-engine/keys/engine_id_rsa").output
+    return {
+        "resource_type": resource_type,
+        "private_key": private_key,
+        "auth_user": auth_user
+    }
+
+
+def __conversion_data_osp(target_provider):  # noqa
+    resource_type = "ManageIQ::Providers::Openstack::CloudManager::Vm"
+    instance_key = conf.credentials[
+        target_provider.data["private-keys"]["conversion_host_ssh_key"]["credentials"]]
+    auth_user = instance_key.username
+    private_key = instance_key.password
+    return {
+        "resource_type": resource_type,
+        "private_key": private_key,
+        "auth_user": auth_user
+    }
+
+
+def set_conversion_host_api(
+        appliance, transformation_method, source_provider, target_provider):
+    """Setting conversion host for RHV and OSP provider via REST"""
+    vmware_ssh_private_key = None
+    vmware_vddk_package_url = None
+    conversion_entity = "vms"
+
+    delete_hosts = appliance.ssh_client.run_rails_command(
+        "'MiqTask.delete_all; ConversionHost.delete_all'")
+    if not delete_hosts.success:
+        pytest.skip(
+            "Failed to delete all conversion hosts:".format(delete_hosts.output))
+
+    if target_provider.one_of(OpenStackProvider):
+        conversion_data = __conversion_data_osp(target_provider)
+        try:
+            hosts = target_provider.data["conversion_instances"]
+        except KeyError:
+            pytest.skip("No conversion instance on provider.")
+    else:
+        conversion_data = __conversion_data_rhev(target_provider)
+        try:
+            hosts = target_provider.hosts.all()
+        except KeyError:
+            pytest.skip("No conversion host on provider.")
+
+    if transformation_method == "SSH":
+        vmware_key = conf.credentials[
+            source_provider.data["private-keys"]["vmware-ssh-key"]["credentials"]]
+        vmware_ssh_private_key = vmware_key.password
+    else:
+        vmware_vddk_package_url = __vddk_url()
+
+    for host in hosts:
+        if target_provider.one_of(RHEVMProvider):
+            host = host.name
```

Comment:
optional:
As per my understanding, we are using a host object in this method, better to cover name in L218
```
hosts = [h.name for h in target_provider.hosts.all]
```